### PR TITLE
refactor(core): use standard sign/verify calls in v1

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Previously this code was jumping through self-inflicted hoops because it
was using `tx.build()` instead of `tx.buildIncomplete()`

Issue: BG-34381

20f94e03f
refactor(core): remove unused parameters in v1 verifyInputSignatures

* isBCH: unused
* ignoreKeyIndices: always set to `false`

Issue: BG-34381

01b3026ad
feat(utxo-lib): expose lower-level signature validation methods

We have instances where we need more fine-grained access to signature 
validation results.

Issue: BG-34381

4b14308dd
refactor(core): convert publicKey to Buffer at callsite

The utxolib will remove support for hex-encoded pubkeys

Issue: BG-34381

### CODE REVIEW

**Review by sangaman — APPROVED** [review]
Very nice refactoring, just a couple of questions/notes.

File: modules/core/src/transactionBuilder.ts [hunk 1]
```diff
@@ -940,45 +942,9 @@ exports.signTransaction = function (params) {
...
-          signatures = signatures.filter(sig => !!sig);
-          // Last, override builder's signatures property to an array including previous signatures, if there are any.
-          const builderSignatures = txb.inputs[index].signatures;
-          const nonEmptySignatures = _.remove(builderSignatures, sig => !!sig);
-          signatures.push.apply(signatures, nonEmptySignatures);
-          txb.inputs[index].signatures = signatures;
-        }
-
-      } else {
-        debug('Signing non-segwit input #%d', index);
```
[the remark this answers is not in the document]
Comment (OttoAllmendinger) [reply modules/core/src/transactionBuilder.ts:969]: We previously entered this branch when `witnessScript === undefined`, but there was no reason to branch at all. We can pass `witnessScript` and `txb.sign()` does the right thing if it is `undefined`. 

I think the clarification is only needed when compared to the old code. The single unbranched `txb.sign()` call seems pretty self-explanatory to me.

File: modules/utxo-lib/src/bitgo/signature.ts [hunk 2]
```diff
@@ -145,32 +156,63 @@ export function parseSignatureScript(input: Input): ParsedSignatureScript {
   return { isSegwitInput, inputClassification, signatures, publicKeys, pubScript };
 }
 
+export function parseSignatureScript2Of3(input: Input): ParsedSignatureScript2Of3 {
+  const result = parseSignatureScript(input) as ParsedSignatureScript2Of3;
+
+  if (![script.types.P2WSH, script.types.P2SH, script.types.P2PKH].includes(result.inputClassification)) {
+    throw new Error(`unexpected inputClassification ${result.inputClassification}`);
+  }
+  if (!result.signatures) {
```
[the remark this answers is not in the document]
Comment (OttoAllmendinger) [reply modules/utxo-lib/src/bitgo/signature.ts:183]: I changed the type to `signatures?: [Buffer, Buffer] | [Buffer, Buffer, Buffer]` so now typescript does not let me write `!result.signatures || result.signatures.length === 0` :)

[the remark this answers is not in the document]
Comment (sangaman) [reply modules/utxo-lib/src/bitgo/signature.ts:183]: Ah yes good point.

**Review by sangaman — APPROVED** [review]

File: modules/core/src/transactionBuilder.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sangaman) [reply modules/core/src/transactionBuilder.ts:969]: That's OK with me, you might be right that this only occurred to me because of the old code.

**Pull request merged**